### DISCUSSION TITLE
Label

### DISCUSSION THREAD

**Discussion by bameyrick** [ONSdigital/design-system#40] [Feedback]
Use this place to discuss the [label component](https://ons-design-system.netlify.com/components/label) in the [ONS Design System](https://ons-design-system.netlify.com)

**Comment by bameyrick** [comment]
We should remove the `label--small` variant as this should never be used.

**Comment by jrbarnes9** [comment]
Label `for` attribute should be documented as `required`

**Comment by AlanHONS** [comment]
Requesting an enhancement to Label functionality so that Description contents can include HTML, not just plain text.
Alessio has talked me through a workaround for immediate IDS project need but asked me to request a proper implementation.

Screenshot of one example of the most complex form field description I've been asked to produce (to satisfy UKSA rigid requirements + previous UR improvements) is below.

Which you can see has requirements for being able to force new lines, use bullets and use external link component freely within the text. But that's just one specific example. So freedom to allow HTML output & other design system components within 'description' is my actual desired solution.

![{9AB0FAA0-0A6C-4672-921A-DE33C7F3B290}](https://github.com/user-attachments/assets/5219e80c-012f-4876-bb9f-ab143eed9f33)

**Reply by MagdalenaLarge** [reply]
Hello @AlanHONS ,
Thank you for raising this issue with us. We are currently focusing on the major release, but we will come back to you as soon as possible to discuss the details of your requirement. Thank you for your patience.

**Reply by MagdalenaLarge** [reply]
Hi @AlanHONS, thank you for providing more details about your label requirement. At this time, we are unable to proceed with this request as it is a one-off requirement and is not currently utilized by other users. However, we will keep this in mind and may consider it in the future if we receive more requests for the same requirement.

**Reply by AlanHONS** [reply]
Thanks and understood. Will let you know if/when I come across other more generally applicable needs.
Meantime UKSA restricted content IDS Hub Project Application journey will proceed using workaround method for rich text label descriptions:
`{% from "components/label/_macro.njk" import onsLabel %}
{% from "components/list/_macro.njk" import onsList %}
{% set test %}
{{
    onsList({
        "element": 'ol',
        "itemsList": [
            {
                "text": 'Introduction'
            },
            {
                "text": 'Questionnaire'
            },
            {
                "text": 'Summary'
            }
        ]
    })
}}
{% endset %}
{{
    onsLabel({
        "text": "Label",
        "for": "some-input",
        "description": test | safe
    })
}}`
